{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## [14.09]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### copy hemin core from manual run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir tree_core/hemin/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "cp orthosnake/hemin/tmp/coreogaligned.fasta.raxml.* tree_core/hemin/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "for f in tree_core/hemin/*.*; do newname=$(echo $f | sed 's/coreogaligned.fasta/hemin_core/'); mv $f $newname; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hemin_core.raxml.bestModel\t    hemin_core.raxml.mlTrees\n",
      "hemin_core.raxml.bestTree\t    hemin_core.raxml.rba\n",
      "hemin_core.raxml.bestTreeCollapsed  hemin_core.raxml.reduced.phy\n",
      "hemin_core.raxml.bootstraps\t    hemin_core.raxml.startTree\n",
      "hemin_core.raxml.log\t\t    hemin_core.raxml.support\n"
     ]
    }
   ],
   "source": [
    "ls tree_core/hemin/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### what snake wants?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate snakemake"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "snakemake --cores 100 -rk --use-conda -s Snakefile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## [15.09]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-rw-r-- 1 manolov manolov 608459 Sep 14 18:46 tmp/og_in_region/hemin/hemin.fasta\n",
      "(snakemake) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ll tmp/og_in_region/hemin/hemin.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "snakemake --cores 100 -rk --use-conda -s Snakefile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### snakemake for hemin done (+)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### lets compare trees (+)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### run other regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "snakemake --cores 100 -rk --use-conda -s Snakefile --latency-wait 100 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### snakemake for capsule left done (+)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# [16.09]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### add more regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genomes:  1212\n",
      "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
      "\u001b[33mUpdating job 4 (cat_core).\u001b[0m\n",
      "\u001b[33mUpdating job 5 (og_in_region).\u001b[0m\n",
      "\u001b[33mUpdating job 8 (orthofinder).\u001b[0m\n",
      "\u001b[33mUpdating job 9 (cat_genes).\u001b[0m\n",
      "\u001b[33mUsing shell: /bin/bash\u001b[0m\n",
      "\u001b[33mProvided cores: 100\u001b[0m\n",
      "\u001b[33mRules claiming more threads will be scaled down.\u001b[0m\n",
      "\u001b[33mJob counts:\n",
      "\tcount\tjobs\n",
      "\t1\tall\n",
      "\t1\tog_in_region\n",
      "\t1\tog_in_region_sequence\n",
      "\t1\ttree_for_core\n",
      "\t1\ttree_for_region\n",
      "\t5\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Wed Sep 16 12:26:24 2020]\u001b[0m\n",
      "\u001b[32mrule og_in_region:\n",
      "    input: blast/glyoxilate_wide/GCF_012934785.1_ASM1293478v1_genomic.blast, orthosnake/glyoxilate_wide/faa/GCF_012934785.1_ASM1293478v1_genomic.fasta\n",
      "    output: tmp/og_in_region/glyoxilate_wide/og_list\n",
      "    jobid: 8539\n",
      "    reason: Missing output files: tmp/og_in_region/glyoxilate_wide/og_list\n",
      "    wildcards: region=glyoxilate_wide\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Wed Sep 16 12:26:24 2020]\u001b[0m\n",
      "\u001b[32mrule tree_for_core:\n",
      "    input: orthosnake/glyoxilate_wide/tmp/coreogaligned.fasta\n",
      "    output: tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.bestTree\n",
      "    jobid: 2\n",
      "    reason: Missing output files: tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.bestTree\n",
      "    wildcards: region=glyoxilate_wide\n",
      "    threads: 50\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\u001b[32m[Wed Sep 16 12:26:26 2020]\u001b[0m\n",
      "\u001b[32mFinished job 8539.\u001b[0m\n",
      "\u001b[32m1 of 5 steps (20%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Wed Sep 16 12:26:26 2020]\u001b[0m\n",
      "\u001b[32mrule og_in_region_sequence:\n",
      "    input: tmp/og_in_region/glyoxilate_wide/og_list, orthosnake/glyoxilate_wide/tmp/coreogaligned.fasta\n",
      "    output: tmp/og_in_region/glyoxilate_wide/glyoxilate_wide.fasta\n",
      "    jobid: 3\n",
      "    reason: Missing output files: tmp/og_in_region/glyoxilate_wide/glyoxilate_wide.fasta; Input files updated by another job: tmp/og_in_region/glyoxilate_wide/og_list\n",
      "    wildcards: region=glyoxilate_wide\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[33mActivating conda environment: /data/ps/operon_travel/.snakemake/conda/c5ea06a9\u001b[0m\n",
      "\n",
      "RAxML-NG v. 1.0.0 released on 07.08.2020 by The Exelixis Lab.\n",
      "Developed by: Alexey M. Kozlov and Alexandros Stamatakis.\n",
      "Contributors: Diego Darriba, Tomas Flouri, Benoit Morel, Sarah Lutteropp, Ben Bettisworth.\n",
      "Latest version: https://github.com/amkozlov/raxml-ng\n",
      "Questions/problems/suggestions? Please visit: https://groups.google.com/forum/#!forum/raxml\n",
      "\n",
      "System: AMD EPYC 7502 32-Core Processor, 64 cores, 2003 GB RAM\n",
      "\n",
      "RAxML-NG was called at 16-Sep-2020 12:26:30 as follows:\n",
      "\n",
      "raxml-ng --msa orthosnake/glyoxilate_wide/tmp/coreogaligned.fasta --model GTR+G+I --threads 50 --prefix tree_core/glyoxilate_wide/glyoxilate_wide_core --redo\n",
      "\n",
      "Analysis options:\n",
      "  run mode: ML tree search\n",
      "  start tree(s): random (10) + parsimony (10)\n",
      "  random seed: 1600248390\n",
      "  tip-inner: OFF\n",
      "  pattern compression: ON\n",
      "  per-rate scalers: OFF\n",
      "  site repeats: ON\n",
      "  fast spr radius: AUTO\n",
      "  spr subtree cutoff: 1.000000\n",
      "  branch lengths: proportional (ML estimate, algorithm: NR-FAST)\n",
      "  SIMD kernels: AVX2\n",
      "  parallelization: coarse-grained (auto), PTHREADS (50 threads), thread pinning: OFF\n",
      "\n",
      "WARNING: Running in REDO mode: existing checkpoints are ignored, and all result files will be overwritten!\n",
      "\n",
      "[00:00:00] Reading alignment from file: orthosnake/glyoxilate_wide/tmp/coreogaligned.fasta\n",
      "\u001b[33mActivating conda environment: /data/ps/operon_travel/.snakemake/conda/414dae4b\u001b[0m\n",
      "\u001b[32m[Wed Sep 16 12:26:31 2020]\u001b[0m\n",
      "\u001b[32mFinished job 3.\u001b[0m\n",
      "\u001b[32m2 of 5 steps (40%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Wed Sep 16 12:26:31 2020]\u001b[0m\n",
      "\u001b[32mrule tree_for_region:\n",
      "    input: tmp/og_in_region/glyoxilate_wide/glyoxilate_wide.fasta\n",
      "    output: tree_region/glyoxilate_wide/glyoxilate_wide.raxml.bestTree\n",
      "    jobid: 1\n",
      "    reason: Missing output files: tree_region/glyoxilate_wide/glyoxilate_wide.raxml.bestTree; Input files updated by another job: tmp/og_in_region/glyoxilate_wide/glyoxilate_wide.fasta\n",
      "    wildcards: region=glyoxilate_wide\n",
      "    threads: 50\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "[00:00:02] Loaded alignment with 52 taxa and 2956230 sites\n",
      "\u001b[33mActivating conda environment: /data/ps/operon_travel/.snakemake/conda/c5ea06a9\u001b[0m\n",
      "\n",
      "RAxML-NG v. 1.0.0 released on 07.08.2020 by The Exelixis Lab.\n",
      "Developed by: Alexey M. Kozlov and Alexandros Stamatakis.\n",
      "Contributors: Diego Darriba, Tomas Flouri, Benoit Morel, Sarah Lutteropp, Ben Bettisworth.\n",
      "Latest version: https://github.com/amkozlov/raxml-ng\n",
      "Questions/problems/suggestions? Please visit: https://groups.google.com/forum/#!forum/raxml\n",
      "\n",
      "System: AMD EPYC 7502 32-Core Processor, 64 cores, 2003 GB RAM\n",
      "\n",
      "RAxML-NG was called at 16-Sep-2020 12:26:35 as follows:\n",
      "\n",
      "raxml-ng --msa tmp/og_in_region/glyoxilate_wide/glyoxilate_wide.fasta --model GTR+G+I --threads 50 --prefix tree_region/glyoxilate_wide/glyoxilate_wide\n",
      "\n",
      "Analysis options:\n",
      "  run mode: ML tree search\n",
      "  start tree(s): random (10) + parsimony (10)\n",
      "  random seed: 1600248395\n",
      "  tip-inner: OFF\n",
      "  pattern compression: ON\n",
      "  per-rate scalers: OFF\n",
      "  site repeats: ON\n",
      "  fast spr radius: AUTO\n",
      "  spr subtree cutoff: 1.000000\n",
      "  branch lengths: proportional (ML estimate, algorithm: NR-FAST)\n",
      "  SIMD kernels: AVX2\n",
      "  parallelization: coarse-grained (auto), PTHREADS (50 threads), thread pinning: OFF\n",
      "\n",
      "[00:00:00] Reading alignment from file: tmp/og_in_region/glyoxilate_wide/glyoxilate_wide.fasta\n",
      "[00:00:00] Loaded alignment with 52 taxa and 13311 sites\n",
      "\n",
      "WARNING: Sequences GCF_002764015.1_ASM276401v1_genomic and GCF_002142715.1_ASM214271v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_002764015.1_ASM276401v1_genomic and GCF_002012305.1_ASM201230v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_002764015.1_ASM276401v1_genomic and GCF_001566615.1_ASM156661v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_002764015.1_ASM276401v1_genomic and GCF_003856995.1_ASM385699v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_002764015.1_ASM276401v1_genomic and GCF_002012145.1_ASM201214v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_002764015.1_ASM276401v1_genomic and GCF_001566635.1_ASM156663v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_002764015.1_ASM276401v1_genomic and GCF_002763995.1_ASM276399v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_001021595.1_ASM102159v1_genomic and GCF_002024865.1_ASM202486v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_001021595.1_ASM102159v1_genomic and GCF_000148605.1_ASM14860v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_001021595.1_ASM102159v1_genomic and GCF_000013265.1_ASM1326v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_001021595.1_ASM102159v1_genomic and GCF_001021615.1_APECO18_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_001021595.1_ASM102159v1_genomic and GCF_000800845.1_ASM80084v2_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_001021595.1_ASM102159v1_genomic and GCF_007833255.1_ASM783325v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_001021595.1_ASM102159v1_genomic and GCF_002844685.1_ASM284468v1_genomic are exactly identical!\n",
      "WARNING: Sequences GCF_000284495.1_ASM28449v1_genomic and GCF_000183345.1_ASM18334v1_genomic are exactly identical!\n",
      "WARNING: Duplicate sequences found: 15\n",
      "\n",
      "NOTE: Reduced alignment (with duplicates and gap-only sites/taxa removed) \n",
      "NOTE: was saved to: /data/ps/operon_travel/tree_region/glyoxilate_wide/glyoxilate_wide.raxml.reduced.phy\n",
      "\n",
      "Alignment comprises 1 partitions and 561 patterns\n",
      "\n",
      "Partition 0: noname\n",
      "Model: GTR+FO+I+G4m\n",
      "Alignment sites / patterns: 13311 / 561\n",
      "Gaps: 0.85 %\n",
      "Invariant sites: 94.37 %\n",
      "\n",
      "\n",
      "NOTE: Binary MSA file created: tree_region/glyoxilate_wide/glyoxilate_wide.raxml.rba\n",
      "\n",
      "Parallelization scheme autoconfig: 10 worker(s) x 5 thread(s)\n",
      "\n",
      "Parallel reduction/worker buffer size: 1 KB  / 0 KB\n",
      "\n",
      "[00:00:00] Generating 10 random starting tree(s) with 52 taxa\n",
      "[00:00:00] Generating 10 parsimony starting tree(s) with 52 taxa\n",
      "[00:00:00] Data distribution: max. partitions/sites/weight per thread: 1 / 113 / 1808\n",
      "[00:00:00] Data distribution: max. searches per worker: 2\n",
      "\n",
      "Starting ML tree search with 20 distinct starting trees\n",
      "\n",
      "[00:00:00 -47930.782631] Initial branch length optimization\n",
      "\n",
      "Alignment comprises 1 partitions and 61692 patterns\n",
      "\n",
      "Partition 0: noname\n",
      "Model: GTR+FO+I+G4m\n",
      "Alignment sites / patterns: 2956230 / 61692\n",
      "Gaps: 1.44 %\n",
      "Invariant sites: 93.56 %\n",
      "\n",
      "\n",
      "NOTE: Binary MSA file created: tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.rba\n",
      "\n",
      "Parallelization scheme autoconfig: 5 worker(s) x 10 thread(s)\n",
      "\n",
      "Parallel reduction/worker buffer size: 1 KB  / 0 KB\n",
      "\n",
      "[00:00:05] Generating 10 random starting tree(s) with 52 taxa\n",
      "[00:00:05] Generating 10 parsimony starting tree(s) with 52 taxa\n",
      "[00:00:00 -41274.285538] Model parameter optimization (eps = 10.000000)\n",
      "[00:00:00 -34790.576805] AUTODETECT spr round 1 (radius: 5)\n",
      "[00:00:01 -29722.949056] AUTODETECT spr round 2 (radius: 10)\n",
      "[00:00:01 -27839.880013] AUTODETECT spr round 3 (radius: 15)\n",
      "[00:00:01 -27839.836791] SPR radius for FAST iterations: 10 (autodetect)\n",
      "[00:00:01 -27839.836791] Model parameter optimization (eps = 3.000000)\n",
      "[00:00:01 -27681.061672] FAST spr round 1 (radius: 10)\n",
      "[00:00:01 -27398.261108] FAST spr round 2 (radius: 10)\n",
      "[00:00:01 -27393.657272] FAST spr round 3 (radius: 10)\n",
      "[00:00:02 -27393.657272] Model parameter optimization (eps = 1.000000)\n",
      "[00:00:02 -27391.139606] SLOW spr round 1 (radius: 5)\n",
      "[00:00:08] Data distribution: max. partitions/sites/weight per thread: 1 / 6170 / 98720\n",
      "[00:00:08] Data distribution: max. searches per worker: 4\n",
      "\n",
      "Starting ML tree search with 20 distinct starting trees\n",
      "\n",
      "[00:00:02 -27391.133337] SLOW spr round 2 (radius: 10)\n",
      "[00:00:08 -11151233.138760] Initial branch length optimization\n",
      "[00:00:03 -27366.027767] SLOW spr round 3 (radius: 5)\n",
      "[00:00:04] [worker #9] ML tree search #10, logLikelihood: -27390.691087\n",
      "[00:00:09 -9856811.453935] Model parameter optimization (eps = 10.000000)\n",
      "[00:00:04] [worker #6] ML tree search #7, logLikelihood: -27364.797922\n",
      "[00:00:04] [worker #1] ML tree search #2, logLikelihood: -27390.682835\n",
      "[00:00:04 -27366.027544] SLOW spr round 4 (radius: 10)\n",
      "[00:00:04] [worker #7] ML tree search #8, logLikelihood: -27364.783391\n",
      "[00:00:05] [worker #5] ML tree search #6, logLikelihood: -27364.804155\n",
      "[00:00:05] [worker #4] ML tree search #5, logLikelihood: -27364.787629\n",
      "[00:00:05 -27366.027542] SLOW spr round 5 (radius: 15)\n",
      "[00:00:05] [worker #3] ML tree search #4, logLikelihood: -27375.996677\n",
      "[00:00:05] [worker #8] ML tree search #9, logLikelihood: -27364.784786\n",
      "[00:00:05 -27366.027542] SLOW spr round 6 (radius: 20)\n",
      "[00:00:06 -27366.027542] SLOW spr round 7 (radius: 25)\n",
      "[00:00:06 -27366.027542] Model parameter optimization (eps = 0.100000)\n",
      "\n",
      "[00:00:06] [worker #0] ML tree search #1, logLikelihood: -27364.782604\n",
      "\n",
      "[00:00:06 -37509.872449] Initial branch length optimization\n",
      "[00:00:06 -28553.757949] Model parameter optimization (eps = 10.000000)\n",
      "[00:00:07 -27463.054119] AUTODETECT spr round 1 (radius: 5)\n",
      "[00:00:07] [worker #9] ML tree search #20, logLikelihood: -27376.050341\n",
      "[00:00:07 -27415.069656] AUTODETECT spr round 2 (radius: 10)\n",
      "[00:00:07 -27415.069645] SPR radius for FAST iterations: 5 (autodetect)\n",
      "[00:00:07 -27415.069645] Model parameter optimization (eps = 3.000000)\n",
      "[00:00:07 -27410.125801] FAST spr round 1 (radius: 5)\n",
      "[00:00:07 -27381.905347] FAST spr round 2 (radius: 5)\n",
      "[00:00:08 -27381.904870] Model parameter optimization (eps = 1.000000)\n",
      "[00:00:08] [worker #4] ML tree search #15, logLikelihood: -27397.607263\n",
      "[00:00:08 -27380.455278] SLOW spr round 1 (radius: 5)\n",
      "[00:00:08] [worker #7] ML tree search #18, logLikelihood: -27397.576515\n",
      "[00:00:08] [worker #2] ML tree search #3, logLikelihood: -27375.895052\n",
      "[00:00:08 -27376.229878] SLOW spr round 2 (radius: 5)\n",
      "[00:00:08] [worker #5] ML tree search #16, logLikelihood: -27397.142504\n",
      "[00:00:08] [worker #6] ML tree search #17, logLikelihood: -27375.909355\n",
      "[00:00:08] [worker #1] ML tree search #12, logLikelihood: -27375.914603\n",
      "[00:00:08] [worker #8] ML tree search #19, logLikelihood: -27376.052168\n",
      "[00:00:09 -27376.229447] SLOW spr round 3 (radius: 10)\n",
      "[00:00:09] [worker #3] ML tree search #14, logLikelihood: -27376.087747\n",
      "[00:00:09 -27376.229443] SLOW spr round 4 (radius: 15)\n",
      "[00:00:09 -27376.229443] SLOW spr round 5 (radius: 20)\n",
      "[00:00:10 -27376.229443] SLOW spr round 6 (radius: 25)\n",
      "[00:00:10 -27376.229443] Model parameter optimization (eps = 0.100000)\n",
      "\n",
      "[00:00:10] [worker #0] ML tree search #11, logLikelihood: -27376.045957\n",
      "\n",
      "[00:00:11] [worker #2] ML tree search #13, logLikelihood: -27364.857172\n",
      "\n",
      "Optimized model parameters:\n",
      "\n",
      "   Partition 0: noname\n",
      "   Rate heterogeneity: GAMMA (4 cats, mean),  alpha: 0.680543 (ML),  weights&rates: (0.250000,0.070038) (0.250000,0.354204) (0.250000,0.917439) (0.250000,2.658319) \n",
      "   P-inv (ML): 0.884061\n",
      "   Base frequencies (ML): 0.257446 0.211070 0.251991 0.279493 \n",
      "   Substitution rates (ML): 1.620764 4.165414 1.028738 0.474114 5.960500 1.000000 \n",
      "\n",
      "\n",
      "Final LogLikelihood: -27364.782604\n",
      "\n",
      "AIC score: 54951.565209 / AICc score: 54953.448988 / BIC score: 55783.659619\n",
      "Free parameters (model + branch lengths): 111\n",
      "\n",
      "WARNING: Best ML tree contains 19 near-zero branches!\n",
      "\n",
      "Best ML tree with collapsed near-zero branches saved to: /data/ps/operon_travel/tree_region/glyoxilate_wide/glyoxilate_wide.raxml.bestTreeCollapsed\n",
      "Best ML tree saved to: /data/ps/operon_travel/tree_region/glyoxilate_wide/glyoxilate_wide.raxml.bestTree\n",
      "All ML trees saved to: /data/ps/operon_travel/tree_region/glyoxilate_wide/glyoxilate_wide.raxml.mlTrees\n",
      "Optimized model saved to: /data/ps/operon_travel/tree_region/glyoxilate_wide/glyoxilate_wide.raxml.bestModel\n",
      "\n",
      "Execution log saved to: /data/ps/operon_travel/tree_region/glyoxilate_wide/glyoxilate_wide.raxml.log\n",
      "\n",
      "Analysis started: 16-Sep-2020 12:26:35 / finished: 16-Sep-2020 12:26:47\n",
      "\n",
      "Elapsed time: 11.747 seconds\n",
      "\n",
      "\u001b[32m[Wed Sep 16 12:26:47 2020]\u001b[0m\n",
      "\u001b[32mFinished job 1.\u001b[0m\n",
      "\u001b[32m3 of 5 steps (60%) done\u001b[0m\n",
      "[00:00:25 -8284290.737097] AUTODETECT spr round 1 (radius: 5)\n",
      "[00:00:28 -7010740.178039] AUTODETECT spr round 2 (radius: 10)\n",
      "[00:00:32 -6780108.528640] AUTODETECT spr round 3 (radius: 15)\n",
      "[00:00:35 -6780091.472891] AUTODETECT spr round 4 (radius: 20)\n",
      "[00:00:36 -6780091.438821] SPR radius for FAST iterations: 15 (autodetect)\n",
      "[00:00:36 -6780091.438821] Model parameter optimization (eps = 3.000000)\n",
      "[00:00:40 -6743237.780747] FAST spr round 1 (radius: 15)\n",
      "[00:00:48 -6643001.172868] FAST spr round 2 (radius: 15)\n",
      "[00:00:51 -6642166.388756] FAST spr round 3 (radius: 15)\n",
      "[00:00:55 -6641703.174557] FAST spr round 4 (radius: 15)\n",
      "[00:00:58 -6641703.174548] Model parameter optimization (eps = 1.000000)\n",
      "[00:01:00 -6641168.055627] SLOW spr round 1 (radius: 5)\n",
      "[00:01:14 -6639905.268634] SLOW spr round 2 (radius: 5)\n",
      "[00:01:27 -6639905.268630] SLOW spr round 3 (radius: 10)\n",
      "[00:01:35] [worker #3] ML tree search #4, logLikelihood: -6640973.776257\n",
      "[00:01:40 -6639905.268630] SLOW spr round 4 (radius: 15)\n",
      "[00:01:44] [worker #1] ML tree search #2, logLikelihood: -6639901.795112\n",
      "[00:01:45] [worker #2] ML tree search #3, logLikelihood: -6641038.590011\n",
      "[00:01:48 -6639905.268630] SLOW spr round 5 (radius: 20)\n",
      "[00:01:49 -6639905.268630] SLOW spr round 6 (radius: 25)\n",
      "[00:01:50 -6639905.268630] Model parameter optimization (eps = 0.100000)\n",
      "\n",
      "[00:01:50] [worker #0] ML tree search #1, logLikelihood: -6639904.629744\n",
      "\n",
      "[00:01:50 -11407322.750526] Initial branch length optimization\n",
      "[00:01:51 -10176128.866646] Model parameter optimization (eps = 10.000000)\n",
      "[00:01:53] [worker #4] ML tree search #5, logLikelihood: -6640972.050429\n",
      "[00:02:04 -8515055.518981] AUTODETECT spr round 1 (radius: 5)\n",
      "[00:02:08 -7144749.349935] AUTODETECT spr round 2 (radius: 10)\n",
      "[00:02:14 -6981201.972978] AUTODETECT spr round 3 (radius: 15)\n",
      "[00:02:17 -6980882.980438] AUTODETECT spr round 4 (radius: 20)\n",
      "[00:02:17 -6980882.969489] SPR radius for FAST iterations: 15 (autodetect)\n",
      "[00:02:17 -6980882.969489] Model parameter optimization (eps = 3.000000)\n",
      "[00:02:21 -6938275.967449] FAST spr round 1 (radius: 15)\n",
      "[00:02:31 -6647157.093080] FAST spr round 2 (radius: 15)\n",
      "[00:02:36 -6643096.397128] FAST spr round 3 (radius: 15)\n",
      "[00:02:39 -6643096.396694] Model parameter optimization (eps = 1.000000)\n",
      "[00:02:42 -6641031.124394] SLOW spr round 1 (radius: 5)\n",
      "[00:02:57 -6640597.928369] SLOW spr round 2 (radius: 5)\n",
      "[00:03:11 -6640276.988346] SLOW spr round 3 (radius: 5)\n",
      "[00:03:21] [worker #2] ML tree search #8, logLikelihood: -6640972.683877\n",
      "[00:03:24] [worker #1] ML tree search #7, logLikelihood: -6639902.048374\n",
      "[00:03:26 -6639903.222694] SLOW spr round 4 (radius: 5)\n",
      "[00:03:28] [worker #3] ML tree search #9, logLikelihood: -6639901.613725\n",
      "[00:03:39 -6639903.221898] SLOW spr round 5 (radius: 10)\n",
      "[00:03:52 -6639903.221893] SLOW spr round 6 (radius: 15)\n",
      "[00:03:59 -6639903.221893] SLOW spr round 7 (radius: 20)\n",
      "[00:04:00 -6639903.221893] SLOW spr round 8 (radius: 25)\n",
      "[00:04:01 -6639903.221893] Model parameter optimization (eps = 0.100000)\n",
      "\n",
      "[00:04:01] [worker #0] ML tree search #6, logLikelihood: -6639902.650858\n",
      "\n",
      "[00:04:01 -8864372.899605] Initial branch length optimization\n",
      "[00:04:02 -6986827.378685] Model parameter optimization (eps = 10.000000)\n",
      "[00:04:07 -6641308.554133] AUTODETECT spr round 1 (radius: 5)\n",
      "[00:04:09] [worker #4] ML tree search #10, logLikelihood: -6639901.735807\n",
      "[00:04:10 -6640697.098854] AUTODETECT spr round 2 (radius: 10)\n",
      "[00:04:15 -6640697.093809] SPR radius for FAST iterations: 5 (autodetect)\n",
      "[00:04:15 -6640697.093809] Model parameter optimization (eps = 3.000000)\n",
      "[00:04:16 -6640686.158907] FAST spr round 1 (radius: 5)\n",
      "[00:04:18 -6640686.058294] FAST spr round 2 (radius: 5)\n",
      "[00:04:21 -6640686.056742] Model parameter optimization (eps = 1.000000)\n",
      "[00:04:22 -6640683.229668] SLOW spr round 1 (radius: 5)\n",
      "[00:04:24] [worker #1] ML tree search #12, logLikelihood: -6640357.066506\n",
      "[00:04:31] [worker #2] ML tree search #13, logLikelihood: -6639900.483233\n",
      "[00:04:35 -6640275.740135] SLOW spr round 2 (radius: 5)\n",
      "[00:04:50 -6639901.393978] SLOW spr round 3 (radius: 5)\n",
      "[00:05:04 -6639901.393289] SLOW spr round 4 (radius: 10)\n",
      "[00:05:07] [worker #3] ML tree search #14, logLikelihood: -6639901.827295\n",
      "[00:05:07] [worker #4] ML tree search #15, logLikelihood: -6640562.625257\n",
      "[00:05:16 -6639901.393273] SLOW spr round 5 (radius: 15)\n",
      "[00:05:23 -6639901.393273] SLOW spr round 6 (radius: 20)\n",
      "[00:05:25 -6639901.393273] SLOW spr round 7 (radius: 25)\n",
      "[00:05:25 -6639901.393273] Model parameter optimization (eps = 0.100000)\n",
      "\n",
      "[00:05:26] [worker #0] ML tree search #11, logLikelihood: -6639901.130187\n",
      "\n",
      "[00:05:26 -8867967.202442] Initial branch length optimization\n",
      "[00:05:26 -6988533.638332] Model parameter optimization (eps = 10.000000)\n",
      "[00:05:32 -6642739.424280] AUTODETECT spr round 1 (radius: 5)\n",
      "[00:05:35 -6641279.431245] AUTODETECT spr round 2 (radius: 10)\n",
      "[00:05:40 -6641279.431234] SPR radius for FAST iterations: 5 (autodetect)\n",
      "[00:05:40 -6641279.431234] Model parameter optimization (eps = 3.000000)\n",
      "[00:05:40 -6641263.620308] FAST spr round 1 (radius: 5)\n",
      "[00:05:43 -6641263.613737] Model parameter optimization (eps = 1.000000)\n",
      "[00:05:44 -6641263.609321] SLOW spr round 1 (radius: 5)\n",
      "[00:05:50] [worker #1] ML tree search #17, logLikelihood: -6639900.999884\n",
      "[00:05:57] [worker #2] ML tree search #18, logLikelihood: -6640561.277850\n",
      "[00:05:58 -6639901.574616] SLOW spr round 2 (radius: 5)\n",
      "[00:05:59] [worker #4] ML tree search #20, logLikelihood: -6639901.495195\n",
      "[00:06:12 -6639901.574580] SLOW spr round 3 (radius: 10)\n",
      "[00:06:24 -6639901.574580] SLOW spr round 4 (radius: 15)\n",
      "[00:06:28] [worker #3] ML tree search #19, logLikelihood: -6639901.976956\n",
      "[00:06:32 -6639901.574580] SLOW spr round 5 (radius: 20)\n",
      "[00:06:33 -6639901.574580] SLOW spr round 6 (radius: 25)\n",
      "[00:06:33 -6639901.574580] Model parameter optimization (eps = 0.100000)\n",
      "\n",
      "[00:06:33] [worker #0] ML tree search #16, logLikelihood: -6639901.170896\n",
      "\n",
      "\n",
      "Optimized model parameters:\n",
      "\n",
      "   Partition 0: noname\n",
      "   Rate heterogeneity: GAMMA (4 cats, mean),  alpha: 0.642745 (ML),  weights&rates: (0.250000,0.062027) (0.250000,0.335216) (0.250000,0.901652) (0.250000,2.701105) \n",
      "   P-inv (ML): 0.880178\n",
      "   Base frequencies (ML): 0.234907 0.250000 0.276670 0.238423 \n",
      "   Substitution rates (ML): 1.056881 4.192466 1.156349 0.632897 5.744623 1.000000 \n",
      "\n",
      "\n",
      "Final LogLikelihood: -6639900.483233\n",
      "\n",
      "AIC score: 13280022.966465 / AICc score: 13280022.974876 / BIC score: 13281454.802681\n",
      "Free parameters (model + branch lengths): 111\n",
      "\n",
      "Best ML tree saved to: /data/ps/operon_travel/tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.bestTree\n",
      "All ML trees saved to: /data/ps/operon_travel/tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.mlTrees\n",
      "Optimized model saved to: /data/ps/operon_travel/tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.bestModel\n",
      "\n",
      "Execution log saved to: /data/ps/operon_travel/tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.log\n",
      "\n",
      "Analysis started: 16-Sep-2020 12:26:30 / finished: 16-Sep-2020 12:33:04\n",
      "\n",
      "Elapsed time: 393.848 seconds\n",
      "\n",
      "\u001b[32m[Wed Sep 16 12:33:04 2020]\u001b[0m\n",
      "\u001b[32mFinished job 2.\u001b[0m\n",
      "\u001b[32m4 of 5 steps (80%) done\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Wed Sep 16 12:33:04 2020]\u001b[0m\n",
      "\u001b[32mlocalrule all:\n",
      "    input: tree_region/glyoxilate_wide/glyoxilate_wide.raxml.bestTree, tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.bestTree\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: tree_core/glyoxilate_wide/glyoxilate_wide_core.raxml.bestTree, tree_region/glyoxilate_wide/glyoxilate_wide.raxml.bestTree\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Wed Sep 16 12:33:04 2020]\u001b[0m\n",
      "\u001b[32mFinished job 0.\u001b[0m\n",
      "\u001b[32m5 of 5 steps (100%) done\u001b[0m\n",
      "\u001b[33mComplete log: /data/ps/operon_travel/.snakemake/log/2020-09-16T122601.460740.snakemake.log\u001b[0m\n",
      "(snakemake) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "snakemake --cores 100 -rk --use-conda -s Snakefile --latency-wait 100 "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
